src/hotspot/share/utilities/vmError.cpp

```diff
@@ -749,8 +749,9 @@ void VMError::report(outputStream* st, bool _verbose) {
 
      // printing Java thread stack trace if it is involved in GC crash
      if (_verbose && _thread && (_thread->is_Named_thread())) {
-       JavaThread*  jt = ((NamedThread *)_thread)->processed_thread();
-       if (jt != NULL) {
+       Thread* thread = ((NamedThread *)_thread)->processed_thread();
+       if (thread != NULL && thread->is_Java_thread()) {
+         JavaThread* jt = static_cast<JavaThread*>(thread);
```

Comment:
as_Java_thread() - maybe just search for this in the patch

Reply:
Done.